.github/workflows/ci.yaml

```diff
@@ -0,0 +1,71 @@
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+    branches:
+      # - main
+      # - release-*
```

Comment:
**chris:Let's only run this action on pushes to main**
  once verification stage completed, I include main only

Reply:
Sure that is fine. Upon further thought, let's only run this action when manually triggered for now using the `on: workflow_dispatch`: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#workflow_dispatch. We can consider running these tests more frequently once we have validated that they work on main.

Reply:
The other option we have is to run on every commit to main, but only run the e2e workflow once the `image` workflow has complete. See how we define this ordering in the device plugin: https://github.com/NVIDIA/k8s-device-plugin/blob/main/.github/workflows/e2e.yaml#L17-L24

Reply:
on:
  workflow_run:
    workflows: [image]
    types:
      - completed
    branches:
      - main 
      
      added